usecase/src/test/java/org/mobilitydata/gtfsvalidator/usecase/ValidateStopTimeTimeCombinationTest.java

```diff
@@ -0,0 +1,461 @@
+        when(thirdStopTimeInSequence.getArrivalTime()).thenReturn(1201);
+        when(thirdStopTimeInSequence.getStopSequence()).thenReturn(8);
+        when(thirdStopTimeInSequence.getTripId()).thenReturn("0");
+
+        final ValidationResultRepository mockResultRepo = mock(ValidationResultRepository.class);
+        final Map<Integer, StopTime> mockStopTimeSequence = new TreeMap<>();
+        mockStopTimeSequence.put(2, firstStopTimeInSequence);
+        mockStopTimeSequence.put(5, secondStopTimeInSequence);
+        mockStopTimeSequence.put(8, thirdStopTimeInSequence);
+
+        final TimeUtils mockTimeUtils = mock(TimeUtils.class);
+        when(mockTimeUtils.convertIntegerToHMMSS(710)).thenReturn("710");
+        when(mockTimeUtils.convertIntegerToHMMSS(920)).thenReturn("920");
+        when(mockTimeUtils.convertIntegerToHMMSS(930)).thenReturn("930");
+        when(mockTimeUtils.convertIntegerToHMMSS(820)).thenReturn("820");
+        when(mockTimeUtils.convertIntegerToHMMSS(1201)).thenReturn("1201");
+
+        final ValidateStopTimeTimeCombination underTest = new ValidateStopTimeTimeCombination();
+        underTest.execute(mockResultRepo, mockStopTimeSequence, mockTimeUtils);
+
+        verify(firstStopTimeInSequence, times(2)).getArrivalTime();
+        verify(firstStopTimeInSequence, times(2)).getDepartureTime();
+
+        verify(secondStopTimeInSequence, times(1)).getArrivalTime();
+        verify(secondStopTimeInSequence, times(3)).getDepartureTime();
+
+        verify(thirdStopTimeInSequence, times(2)).getArrivalTime();
+        verify(thirdStopTimeInSequence, times(3)).getDepartureTime();
+        verify(thirdStopTimeInSequence, times(1)).getTripId();
+        verify(thirdStopTimeInSequence, times(1)).getStopSequence();
+
+        final ArgumentCaptor<BadStopTimeTimeCombinationNotice> captor =
+                ArgumentCaptor.forClass(BadStopTimeTimeCombinationNotice.class);
+
+        verify(mockResultRepo, times(1)).addNotice(captor.capture());
+
+        BadStopTimeTimeCombinationNotice notice = captor.getValue();
+        assertEquals("stop_times.txt", notice.getFilename());
+        assertEquals("tripId", notice.getNoticeSpecific(KEY_COMPOSITE_KEY_FIRST_PART));
+        assertEquals("stopSequence", notice.getNoticeSpecific(KEY_COMPOSITE_KEY_SECOND_PART));
+        assertEquals("0", notice.getNoticeSpecific(KEY_COMPOSITE_KEY_FIRST_VALUE));
+        assertEquals(8, notice.getNoticeSpecific(KEY_COMPOSITE_KEY_SECOND_VALUE));
+        assertEquals("930", notice.getNoticeSpecific(KEY_STOP_TIME_ARRIVAL_TIME));
+        assertEquals("820", notice.getNoticeSpecific(KEY_STOP_TIME_DEPARTURE_TIME));
+        assertEquals(5, notice.getNoticeSpecific(KEY_STOP_TIME_STOP_SEQUENCE));
+        assertEquals("no id", notice.getEntityId());
+
+        verifyNoMoreInteractions(firstStopTimeInSequence, secondStopTimeInSequence, thirdStopTimeInSequence,
+                mockResultRepo);
+    }
```

Comment:
Looks like you're missing the test case here for when there is backwards time travel and the middle rows are null:

 | tripId | stopSequence | arrivalTime | departureTime |
 |--------|--------------|---------------|-------------|
 | 0      | 2            | 07:10         | 09:20       |
 | 0      | 5            | null                 | null        |
 | 0      | 8            | 08:20         | 12:01       |

Reply:
Good catch, thanks! :)